modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotChecker.java

```diff
@@ -872,4 +1277,78 @@ private SnapshotEncryptionKeyProvider(GridKernalContext ctx, Map<Integer, File>
             return key != null && key.id() == keyId ? key : null;
         }
     }
+
+    /**  */
+    public static class IncrementalSnapshotResult implements Serializable {
```

Comment:
public static nested classes are bad practice. It looks like we can just use existing class IncrementalSnapshotVerificationTaskResult.

Reply:
We are removing the tasks (and related classes) in a next ticket. There will be no IncrementalSnapshotVerificationTaskResult. It is created only within the checker, has a private ctor.

Reply:
Let's create a new high level class instead of nested then. Public static classes look bad.